worker/src/handlers/aggregationProcessor/aggregationProcessor.js

```diff
@@ -365,7 +403,17 @@ const aggregationProcessor = async (
       fromTimestamp: fromTimestamp.toDate(),
       toTimestamp: toTimestamp.toDate(),
       greaterThanDate: model.greaterThanDate,
-      results
+      results,
+      lastRun: actualNow.toDate(),
+      lastCompletedRun: hasReachedEnd({
+        model: {
+          ...model,
+          toTimestamp: toTimestamp.toDate(),
+          fromTimestamp: fromTimestamp.toDate()
+        },
+        now
+      }) ?
+        actualNow.toDate() : model.lastCompletedRun
```

Comment:
I think this code would be easier to read if the result of `hasReachedEnd` was assigned to a variable above the `newModel` definition.